hooks/queries/tokenMetadata.ts

```diff
@@ -0,0 +1,40 @@
+import { findMetadataPda } from '@metaplex-foundation/js'
+import { Metadata } from '@metaplex-foundation/mpl-token-metadata'
+import { PublicKey } from '@solana/web3.js'
+import { useQuery } from '@tanstack/react-query'
+import useWalletStore from 'stores/useWalletStore'
+
+export const useTokenMetadata = (
+  mint: PublicKey | undefined,
+  enableConditions = true
+) => {
+  const connection = useWalletStore((s) => s.connection)
+
+  const enabled = !!mint && !!enableConditions
+
+  const query = useQuery({
+    queryFn: async () => {
+      const mintPubkey = new PublicKey(mint!)
+      const metadataAccount = findMetadataPda(mintPubkey)
+      const accountData = await connection.current.getAccountInfo(
+        metadataAccount
+      )
+
+      const state = Metadata.deserialize(accountData!.data)
+      const jsonUri = state[0].data.uri.slice(
```

Comment:
😮 what about just using `Metadata.fromAccountAddress` and then just `result.uri` or whatever? this looks harder to maintain in future